File: VB/DeleteAllListViewItems.Module.Win/DeleteAllItemsControllerWin.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base
Imports System.Collections
Imports DevExpress.ExpressApp.Win.Editors
Imports DevExpress.Data.Filtering

Namespace DeleteAllListViewItems.Module
	Partial Public Class DeleteAllItemsController
		Inherits ViewController
		Public Sub New()
			InitializeComponent()
			RegisterActions(components)
			Me.TargetViewType = ViewType.ListView
		End Sub


		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()
			UpdateActionState()
			AddHandler View.SelectionChanged, AddressOf ObjectChanged
			AddHandler View.CurrentObjectChanged, AddressOf ObjectChanged
		End Sub

		Private Sub ObjectChanged(ByVal sender As Object, ByVal e As EventArgs)
			UpdateActionState()
		End Sub


		Private Sub UpdateActionState()
			Dim type As Type = (CType(View, ListView)).ObjectTypeInfo.Type
			Me.simpleAction1.Enabled.SetItemValue("ItemsCount", View.ObjectSpace.GetObjects(type).Count > 0)
		End Sub


		Private Sub simpleAction1_Execute(ByVal sender As Object, ByVal e As SimpleActionExecuteEventArgs) Handles simpleAction1.Execute
			Dim editor As GridListEditor = TryCast((TryCast(View, ListView)).Editor, GridListEditor)
			If editor IsNot Nothing Then
				Using objectSpace As IObjectSpace = Application.CreateObjectSpace()
					Dim objectsToDelete As New ArrayList()
					Dim type As Type = (CType(View, ListView)).ObjectTypeInfo.Type
					objectsToDelete.AddRange(objectSpace.GetObjects(type, editor.GridView.ActiveFilterCriteria))
					objectSpace.Delete(objectsToDelete)
					objectSpace.CommitChanges()
				End Using
				ObjectSpace.Refresh()
				UpdateActionState()
			End If
		End Sub
	End Class
End Namespace

File: CS/DeleteAllListViewItems.Module.Win/DeleteAllItemsControllerWin.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using System.Collections;
using DevExpress.ExpressApp.Win.Editors;
using DevExpress.Data.Filtering;

namespace DeleteAllListViewItems.Module
{
    public partial class DeleteAllItemsController : ViewController
    {
        public DeleteAllItemsController()
        {
            InitializeComponent();
            RegisterActions(components);
            this.TargetViewType = ViewType.ListView;
        }


        protected override void OnActivated()
        {
            base.OnActivated();
            UpdateActionState();
            View.SelectionChanged += new EventHandler(ObjectChanged);
            View.CurrentObjectChanged += new EventHandler(ObjectChanged);
        }

        void ObjectChanged(object sender, EventArgs e)
        {
            UpdateActionState();
        }


        private void UpdateActionState()
        {
            Type type = ((ListView)View).ObjectTypeInfo.Type;
            this.simpleAction1.Enabled.SetItemValue("ItemsCount", View.ObjectSpace.GetObjects(type).Count > 0);
        }


        private void simpleAction1_Execute(object sender, SimpleActionExecuteEventArgs e)
        {
            GridListEditor editor = (View as ListView).Editor as GridListEditor;
            if (editor != null)
            {
                using (IObjectSpace objectSpace = Application.CreateObjectSpace())
                {
                    ArrayList objectsToDelete = new ArrayList();
                    Type type = ((ListView)View).ObjectTypeInfo.Type;
                    objectsToDelete.AddRange(objectSpace.GetObjects(type, editor.GridView.ActiveFilterCriteria));
                    objectSpace.Delete(objectsToDelete);
                    objectSpace.CommitChanges();
                }
                ObjectSpace.Refresh();
                UpdateActionState();
            }
        }
    }
}
